File: php/MObjects/GetMObjects.php
<?php
$debug = true;
$marketoSoapEndPoint    = "";  // CHANGE ME
$marketoUserId      = "";  // CHANGE ME
$marketoSecretKey   = "";  // CHANGE ME
$marketoNameSpace   = "http://www.marketo.com/mktows/";
 
// Create Signature
$dtzObj = new DateTimeZone("America/Los_Angeles");
$dtObj  = new DateTime('now', $dtzObj);
$timeStamp = $dtObj->format(DATE_W3C);
$encryptString = $timeStamp . $marketoUserId;
$signature = hash_hmac('sha1', $encryptString, $marketoSecretKey);
 
// Create SOAP Header
$attrs = new stdClass();
$attrs->mktowsUserId = $marketoUserId;
$attrs->requestSignature = $signature;
$attrs->requestTimestamp = $timeStamp;
$authHdr = new SoapHeader($marketoNameSpace, 'AuthenticationHeader', $attrs);
$options = array("connection_timeout" => 15, "location" => $marketoSoapEndPoint);
if ($debug) {
  $options["trace"] = 1;
}
// Create Request
$params->type = 'Program';
// $params->id = "1003";
 
$mObjCriteria1 = new stdClass();
$mObjCriteria1->attrName="Id";
$mObjCriteria1->comparison="LE";
$mObjCriteria1->attrValue="1010";
 
$mObjCriteria2 = new stdClass();
$mObjCriteria2->attrName="Name";
$mObjCriteria2->comparison="NE";
$mObjCriteria2->attrValue="elizprogramtest";
 
$params->mObjCriteriaList=array($mObjCriteria1, $mObjCriteria2);
 
$soapClient = new SoapClient($marketoSoapEndPoint ."?WSDL", $options);
try {
  $leads = $soapClient->__soapCall('getMObjects', array($params), $options, $authHdr);
  //      print_r($leads);
}
catch(Exception $ex) {
  var_dump($ex);
}
if ($debug) {
  print "RAW request:\n" .$soapClient->__getLastRequest() ."\n";
  print "RAW response:\n" .$soapClient->__getLastResponse() ."\n";
}
?>

File: ruby/MObjects/GetMObjects.rb
#
# Marketo SOAP API Sample Code
# Copyright (C) 2016 Marketo, Inc.
#
# This software may be modified and distributed under the terms
# of the MIT license.  See the LICENSE file for details.
#
require 'savon' # Use version 2.0 Savon gem
require 'date'

mktowsUserId = "" # CHANGE ME
marketoSecretKey = "" # CHANGE ME
marketoSoapEndPoint = "" # CHANGE ME
marketoNameSpace = "http://www.marketo.com/mktows/"

#Create Signature
Timestamp = DateTime.now
requestTimestamp = Timestamp.to_s
encryptString = requestTimestamp + mktowsUserId
digest = OpenSSL::Digest.new('sha1')
hashedsignature = OpenSSL::HMAC.hexdigest(digest, marketoSecretKey, encryptString)
requestSignature = hashedsignature.to_s

#Create SOAP Header
headers = { 
	'ns1:AuthenticationHeader' =&gt; { "mktowsUserId" =&gt; mktowsUserId, "requestSignature" =&gt; requestSignature, 					
	"requestTimestamp"  =&gt; requestTimestamp 
	}
}

client = Savon.client(wsdl: 'http://app.marketo.com/soap/mktows/2_3?WSDL', soap_header: headers, endpoint: marketoSoapEndPoint, open_timeout: 90, read_timeout: 90, namespace_identifier: :ns1, env_namespace: 'SOAP-ENV')

#Create Request
request = {
	:type =&gt; "Program",
	:m_obj_criteria_list =&gt; {
  		:m_obj_criteria =&gt; {
    		:attr_name =&gt; "Id",
    		:comparsion =&gt; "LE",
    		:attr_value =&gt; "1010"
  		},
  		:m_obj_criteria! =&gt; {
    		:attr_name =&gt; "Name",
    		:comparsion =&gt; "NE",
    		:attr_value =&gt; "elizprogramtest"
  		}
  	}
}

response = client.call(:get_m_objects, message: request)

puts response